### DISCUSSION TITLE
Is there an API document?

### DISCUSSION THREAD

**Discussion by FoyonaCZY** [usememos/memos#3045] [Q&A] [answered]
I want to creat an app of Memos in flutter,but I can't find an official or unofficial API document so I don't know how to get the datas.

**Comment by DipsySu** [comment]
you can read the API documentation here [swagger](https://github.com/usememos/memos/blob/main/server/route/api/v1/swagger.md)

**Reply by wsw70** [reply]
This link is gone, the file does not exist and it is (now?) `router` with an "r"

**Comment by lincolnthalles** [comment, marked as answer]
Prefer the newer API.

https://memos.apidocumentation.com/reference

https://github.com/usememos/memos/blob/main/proto/gen/apidocs.swagger.yaml

~~https://github.com/usememos/memos/blob/main/docs/apidocs.swagger.yaml~~

~~https://github.com/usememos/memos/blob/main/server/route/api/v2/apidocs.swagger.md~~

~~https://github.com/usememos/memos/blob/main/server/route/api/v2/apidocs.swagger.yaml~~

**Reply by FoyonaCZY** [reply]
Thanks a lot

**Reply by wsw70** [reply]
These URLs are 404s, I could just find `/server/router/api/v1` - route**r** (with an "r") and only v1

**Reply by lincolnthalles** [reply]
> These URLs are 404s, I could just find `/server/router/api/v1` - route**r** (with an "r") and only v1

Those files were moved. You can find the current API documentation here:

https://memos.apidocumentation.com/reference

https://github.com/usememos/memos/blob/main/docs/apidocs.swagger.yaml

**Comment by wsw70** [comment]
There is something in https://github.com/orgs/usememos/discussions/1024 but it is in an Asian language (Chinese probably). It is at least a start.

**Comment by KhalfaniW** [comment]
https://github.com/usememos/memos/blob/v0.20.0/api/v2/apidocs.swagger.md

There was an api document that was removed in v0.20.1 

I'm not sure how much of this is relevant
